kratos/geometries/brep_surface.h

```diff
@@ -594,6 +612,10 @@ class BrepSurface
 
     BrepCurveOnSurfaceArrayType mEmbeddedEdgesArray;
 
+
+    GeometrySurrogateArrayType mpSurrogateInnerLoopGeometries;
+    GeometrySurrogateArrayType mpSurrogateOuterLoopGeometries;
+
```

Comment:
You are adding two member variables but the copy constructor and the assignment operator do not take these into account. Furthermore, these are not added to the serialization so you'll be breaking the restart mechanism.